### ISSUE TITLE
validatePassword not working with RAR5

### ISSUE THREAD

**Issue by SkiesOFF** [abbeycode/UnrarKit#19]
Hello,

The method validatePassword don't work if the archive is RAR5.

You can test using this archive (password is 123):
https://www.dropbox.com/s/rnl6y9jons1eiyp/rar5WithPassword.rar?dl=0

This happens because this if

```objective-c
if (RHCode == ERAR_MISSING_PASSWORD || PFCode == ERAR_MISSING_PASSWORD || RHCode == ERAR_BAD_DATA || PFCode == ERAR_BAD_DATA)
            return;
```

is not checking if PFCode is ERAR_BAD_PASSWORD, which is the value a rar5 with wrong password returns. So, change this if to ->

```objective-c
if (RHCode == ERAR_MISSING_PASSWORD || PFCode == ERAR_BAD_PASSWORD || PFCode == ERAR_MISSING_PASSWORD || RHCode == ERAR_BAD_DATA || PFCode == ERAR_BAD_DATA)
            return;
```

to fix this issue =).

**Comment by abbeycode** [comment]
Thanks, I'll take a look at writing a test case with a RAR5 file to validate your proposed change.

**Comment by abbeycode** [comment]
Fixed, and released to CocoaPods as v2.4.2
